### DISCUSSION TITLE
Fields in Settings are duplicated

### DISCUSSION THREAD

**Discussion by wodka** [directus/directus#10557] [Archive]
### Describe the Bug

When opening the settings the fields after the first in "Branding & Style" and "Files & Storage" are multiplied

![image](https://user-images.githubusercontent.com/385731/146340014-7fc52029-58ad-4502-b133-e9aeb7f86eb6.png)

Info Api returns: 
![image](https://user-images.githubusercontent.com/385731/146342027-119e3409-01e8-4ae5-ae27-cf0f3287f74a.png)

### To Reproduce

unclear - same deployment does not produce this result locally.

in the fields api request there are multiple entries for the duplicated fields. Why would it create new ones there?

![image](https://user-images.githubusercontent.com/385731/146342409-202bda36-0e8c-4e06-8ca7-60b4f4c026f1.png)

### Errors Shown

there are no errors shown / and no errors in the chrome console

### What version of Directus are you using?

9.2.2

### What version of Node.js are you using?

16.13.1

### What database are you using?

CockroachDB

### What browser are you using?

Chrome

### What operating system are you using?

OSX Monterey

### How are you deploying Directus?

Docker

**Comment by azrikahar** [comment]
I'm curious whether this is something that will only happen in CockroachDB (Thanks for the PR #10113 btw!) as this does look rather bizarre. Just curious, by "same deployment", do you mean both the "production" and local deployment are the exact same? Does it happen again when you redeploy it (not locally)?

That said, it would certainly qualify for as a bug with Directus if this happens to the current DBs officially supported by Directus.

**Comment by wodka** [comment]
with same deployment I mean that its running the same code. (I replaced the directus files in node modules with the ones from lerna build to have cockroach support with all migrations). The files are copied into a docker image to then be able to run directus start - and I'm using the same image both on the Staging as well as my local system.

Strangely enough it does not happen locally even if I start with a new database.

I'll later on try with a new database on our staging system

**Comment by azrikahar** [comment]
> (I replaced the directus files in node modules with the ones from lerna build to have cockroach support with all migrations)

Oh that makes sense. Perhaps it might be related to this after all. 

Currently the system fields are located over here in form of yaml files: https://github.com/directus/directus/tree/main/api/src/database/system-data/fields

And then later on read & grabbed by the API into `systemFieldRows`:

https://github.com/directus/directus/blob/741e62827430adbd72c3255914ef55a48bc8ea32/api/src/database/system-data/fields/index.ts#L14-L37

which then gets pushed to the array of fields in the fields API request:

https://github.com/directus/directus/blob/741e62827430adbd72c3255914ef55a48bc8ea32/api/src/services/fields.ts#L75

Maybe one way to check would be to verify whether the YAML files are somehow "built wrongly" (also not super sure how this is even possible) which may explain why it ended up adding the system fields weirdly.

**Comment by wodka** [comment]
mhmm :/ just trying to figure out more there. thanks for links how how this is being created - will add a few log messages in there to see what is happening on that system.

The sort has always the same number -> so the yaml files themself are ok. Even weirder is that not all fields are duplicated...

![image](https://user-images.githubusercontent.com/385731/146358120-ae2484e2-1535-4fbe-8c51-a1474fe3c777.png)

**Comment by wodka** [comment]
just realised that it is not limited to settings

![image](https://user-images.githubusercontent.com/385731/146358980-93a252e3-d8b4-4e39-a4bd-bc729513406f.png)

and that all of them are repeated 18 times^^

```
directus_collections/group 18
directus_dashboards/user_created 18
directus_files/folder 18
directus_files/uploaded_by 18
directus_files/modified_by 18
directus_folders/parent 18
directus_notifications/recipient 18
directus_notifications/sender 18
directus_panels/dashboard 18
directus_panels/user_created 18
directus_permissions/role 18
directus_presets/user 18
directus_presets/role 18
directus_revisions/activity 18
directus_revisions/parent 18
directus_sessions/user 18
directus_settings/project_logo 18
directus_settings/public_foreground 18
directus_settings/public_background 18
directus_settings/storage_default_folder 18
directus_users/role 18
```

with other 195 fields only being returned only once (as they should)

**Comment by wodka** [comment]
HA!

it does return the relation for several entries in schema - so they actually are not identical results!

```json
{
  "data": [
	{
	  "collection": "directus_settings",
	  "field": "project_logo",
	  "type": "uuid",
	  "schema": {
		"...": "...",
		"foreign_key_schema": "public",
		"foreign_key_table": "directus_presets",
		"foreign_key_column": "id"
	  },
	  "meta": "..."
	},
	{
	  "collection": "directus_settings",
	  "field": "project_logo",
	  "type": "uuid",
	  "schema": {
		"...": "...",
		"foreign_key_schema": "public",
		"foreign_key_table": "directus_notifications",
		"foreign_key_column": "id"
	  },
	  "meta": "..."
	},
	{
	  "collection": "directus_settings",
	  "field": "project_logo",
	  "type": "uuid",
	  "schema": {
		"...": "...",
		"foreign_key_schema": "public",
		"foreign_key_table": "directus_panels",
		"foreign_key_column": "id"
	  },
	  "meta": "..."
	},
	"..."
```

thinking now what might have gone wrong there... for reference Locally I'm running CRDB 21.1.10 and on the Staging System it is 21.2.10

So next for me is updating and checking if the error then starts to happen locally as well

**Comment by wodka** [comment]
ok, it does not make a difference - but it is showing the schema entry for every table in the database.

-> just confirmed that on the staging system for me it duplicates those fields for every table in the database. (still unclear why)

**Comment by rijkvanzanten** [comment]
I'll move this to a discussion for now, as CockroachDB is technically not officially supported yet. We can move it back in here if it turns out to be a bug in Directus itself that affects the supported database vendors 👍🏻

**Comment by wodka** [comment]
ok, problem comes from naming the constraint relation names in cockroach:

![image](https://user-images.githubusercontent.com/385731/146411393-537d07ed-77ba-4f3d-af1e-a9af93e821c1.png)

**Reply by wodka** [reply]
and on my local system they are all named differently...

**Comment by wodka** [comment]
the whole problem is coming up because the db structure that I imported from DataGrip changed all names for Primary keys to "primary"...

So I think this is a none issue if the structure is created through the migrations - that is way it was working correctly there.

Sorry for the huge confusion - it is actually a non issue!

**Reply by rijkvanzanten** [reply]
Gotcha! Thanks for confirming that 🙂
